Arch Linux compilation issue

```bash

pkgbase=bin2cpp
pkgname=bin2cpp
pkgver=3.1.0
pkgrel=0
pkgdesc="bin2cpp: The easiest way to embed small files into a c++ executable. bin2cpp converts text or binary files to C++ files (*.h, *.cpp) for easy access within the code."
arch=($CARCH)
url="https://github.com/end2endzone/bin2cpp"
license=('MIT')
groups=()
provides=(${pkgname})
conflicts=(${pkgname})
replaces=()
depends=(
    gcc-libs
    glibc
)
makedepends=(
    cmake
    doxygen
    gtest
    git
    ninja
    pkgconf
    #AUR
    rapidassist
)
checkdepends=()
optdepends=()
source=("${pkgname}::git+${url}.git#tag=${pkgver}")
sha256sums=('511965596926455f0c73c20b43a2b5264e3456ba057b373c09b27ab9f2bb1c2f')
options=(!strip !lto !debug)

build() {
    PLATFORM=x64

    cd "${srcdir}/${pkgname}/"

    sed -e 's|"."|"share/bin2cpp"|g' \
        -e 's|"docs"|"share/doc/bin2cpp"|g' \
        -i CMakeLists.txt
    # see：https://wiki.archlinux.org/title/CMake_package_guidelines
    cmake -DCMAKE_BUILD_TYPE=Release \
        -DCMAKE_INSTALL_PREFIX=/usr \
        -DBIN2CPP_BUILD_TEST=OFF \
        -DBIN2CPP_BUILD_SAMPLES=OFF \
        -DBUILD_SHARED_LIBS=ON \
        -DCMAKE_POLICY_VERSION_MINIMUM=3.5 \
        -Wno-dev \
        -B build \
        -G Ninja

    ninja -C build
}

package() {
    DESTDIR="${pkgdir}" ninja -C "${srcdir}"/${pkgname}/build install
    install -Dm0644 "${srcdir}"/${pkgname}/LICENSE -t ${pkgdir}/usr/share/licenses/${pkgname}/
}

```


```bash
resolving dependencies...
looking for conflicting packages...

Package (21)              New Version     Net Change

extra/clang               20.1.8-1        210.15 MiB
extra/compiler-rt         20.1.8-1         45.59 MiB
extra/cppdap              1.58.0-2          1.55 MiB
extra/fmt                 11.2.0-1          0.69 MiB
extra/hicolor-icon-theme  0.18-1            0.05 MiB
extra/jsoncpp             1.9.6-3           0.77 MiB
core/libedit              20250104_3.1-1    0.26 MiB
extra/libuv               1.51.0-1          0.62 MiB
extra/llvm-libs           20.1.8-1        150.01 MiB
extra/perl-error          0.17030-2         0.04 MiB
extra/perl-mailtools      2.22-2            0.10 MiB
extra/perl-timedate       2.33-8            0.08 MiB
extra/rhash               1.4.4-1           0.32 MiB
extra/spdlog              1.15.3-1          0.68 MiB
extra/zlib-ng             2.2.5-1           0.26 MiB
extra/cmake               4.0.3-1          85.01 MiB
extra/doxygen             1.14.0-1         19.24 MiB
extra/git                 2.50.1-3         29.40 MiB
extra/gtest               1.17.0-1          1.63 MiB
extra/ninja               1.12.1-2          0.38 MiB
aur-repo/rapidassist      0.11.0-2          0.72 MiB

Total Installed Size:  547.53 MiB

:: Proceed with installation? [Y/n]
checking keyring...
checking package integrity...
loading package files...
checking for file conflicts...
:: Processing package changes...
installing cppdap...
installing hicolor-icon-theme...
installing jsoncpp...
Optional dependencies for jsoncpp
jsoncpp-doc: documentation
installing libuv...
installing rhash...
installing cmake...
Optional dependencies for cmake
make: for unix Makefile generator [installed]
ninja: for ninja generator [pending]
qt6-base: cmake-gui
installing libedit...
installing llvm-libs...
installing compiler-rt...
installing clang...
Optional dependencies for clang
openmp: OpenMP support in clang with -fopenmp
python: for scan-view and git-clang-format
llvm: referenced by some clang headers
installing fmt...
installing spdlog...
installing doxygen...
Optional dependencies for doxygen
graphviz: for caller/callee graph generation
qt6-base: for doxywizard
texlive-fontsrecommended: for generating LaTeX, Postscript and PDF output
texlive-fontutils: for generating LaTeX, Postscript and PDF output
texlive-latexextra: for generating LaTeX, Postscript and PDF output
texlive-plaingeneric: for generating LaTeX, Postscript and PDF output
installing gtest...
Optional dependencies for gtest
python: gmock generator
installing perl-error...
installing perl-timedate...
installing perl-mailtools...
installing zlib-ng...
installing git...
Optional dependencies for git
git-zsh-completion: upstream zsh completion
tk: gitk and git gui
openssh: ssh transport and crypto
man: show help with `git command --help`
perl-libwww: git svn
perl-term-readkey: git svn and interactive.singlekey setting
perl-io-socket-ssl: git send-email TLS support
perl-authen-sasl: git send-email TLS support
perl-mediawiki-api: git mediawiki support
perl-datetime-format-iso8601: git mediawiki support
perl-lwp-protocol-https: git mediawiki https support
perl-cgi: gitweb (web interface) support
python: git svn & git p4
subversion: git svn
org.freedesktop.secrets: keyring credential helper
libsecret: libsecret credential helper [installed]
less: the default pager for git
installing ninja...
installing rapidassist...
:: Running post-transaction hooks...
(1/4) Creating system user accounts...
Creating group 'git' with GID 972.
Creating user 'git' (git daemon user) with UID 972 and GID 972.
(2/4) Reloading system manager configuration...
Skipped: Current root is not booted.
(3/4) Arming ConditionNeedsUpdate...
(4/4) Checking for old perl modules...
==> Retrieving sources...
==> WARNING: Skipping all source file integrity checks.
==> Extracting sources...
-> Creating working copy of bin2cpp git repo...
Cloning into 'bin2cpp'...
done.
Switched to a new branch 'makepkg'
==> Starting build()...
-- The C compiler identification is GNU 15.2.1
-- The CXX compiler identification is GNU 15.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Could NOT find WIX (missing: WIX_CANDLE WIX_LIGHT WIX_ROOT_PATH WIX_BIN_DIR)
-- Could NOT find NSIS (missing: NSIS_MAKE)
-- PROJECT_NAME:                bin2cpp
-- CMAKE_BINARY_DIR:            /build/bin2cpp/src/bin2cpp/build
-- CMAKE_CURRENT_BINARY_DIR:    /build/bin2cpp/src/bin2cpp/build
-- CMAKE_SOURCE_DIR:            /build/bin2cpp/src/bin2cpp
-- CMAKE_CURRENT_SOURCE_DIR:    /build/bin2cpp/src/bin2cpp
-- PROJECT_BINARY_DIR:          /build/bin2cpp/src/bin2cpp/build
-- PROJECT_SOURCE_DIR:          /build/bin2cpp/src/bin2cpp
-- EXECUTABLE_OUTPUT_PATH:      /build/bin2cpp/src/bin2cpp/build/bin
-- LIBRARY_OUTPUT_PATH:         /build/bin2cpp/src/bin2cpp/build/bin
-- CMAKE_MODULE_PATH:           /build/bin2cpp/src/bin2cpp/cmake
-- CMAKE_COMMAND:               /usr/bin/cmake
-- CMAKE_ROOT:                  /usr/share/cmake
-- CMAKE_CURRENT_LIST_FILE:     /build/bin2cpp/src/bin2cpp/CMakeLists.txt
-- CMAKE_CURRENT_LIST_LINE:     44
-- CMAKE_INCLUDE_PATH:
-- CMAKE_LIBRARY_PATH:
-- CMAKE_SYSTEM:                Linux-6.16.0-zen2-1-zen
-- CMAKE_SYSTEM_NAME:           Linux
-- CMAKE_SYSTEM_VERSION:        6.16.0-zen2-1-zen
-- CMAKE_SYSTEM_PROCESSOR:      x86_64
-- CMAKE_HOST_SYSTEM_PROCESSOR: x86_64
-- CMAKE_SYSTEM_INFO_FILE:      Platform/Linux
CMake Error at CMakeLists.txt:86 (get_major_version):
get_major_version Function invoked with incorrect arguments for function
named: get_major_version


-- LINUX_DIST_NAME:             arch
-- LINUX_DIST_VER:
-- LINUX_DIST_VER_MAJOR:
Generating /build/bin2cpp/src/bin2cpp/build/include/bin2cpp/version.h...
Generating /build/bin2cpp/src/bin2cpp/build/include/bin2cpp/config.h...
-- Found GTest: /usr/lib/cmake/GTest/GTestConfig.cmake (found version "1.17.0")
-- Configuring incomplete, errors occurred!
==> ERROR: A failure occurred in build().
Aborting...

```

--- a/CHANGES
+++ b/CHANGES
@@ -1,3 +1,7 @@
+Changes for 3.1.1:
+
+* Fixed issue #82: Use the kernel version as a fallback
+
 Changes for 3.1.0:
 
 * New feature: Support for generating C code.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -82,8 +82,15 @@ elseif (UNIX)
   # Get Linux distribution info.
   GetOsReleaseInfo(LINUX_DIST_NAME LINUX_DIST_VER )
 
+  if("${LINUX_DIST_VER}" STREQUAL "")
+    # Use the kernel version as a fallback
+    execute_process(COMMAND uname -r
+      OUTPUT_VARIABLE LINUX_DIST_VER
+      OUTPUT_STRIP_TRAILING_WHITESPACE)
+  endif()
+
   # Extract distribution MAJOR version
-  get_major_version(${LINUX_DIST_VER} LINUX_DIST_VER_MAJOR)
+  get_major_version("${LINUX_DIST_VER}" LINUX_DIST_VER_MAJOR)
 
   MESSAGE( STATUS "LINUX_DIST_NAME:             " ${LINUX_DIST_NAME} )
   MESSAGE( STATUS "LINUX_DIST_VER:              " ${LINUX_DIST_VER} )

